File: 백준/Silver/1931. 회의실 배정/회의실 배정.py
n = int(input())
meetings = list()
for i in range(n):
    meetings.append(tuple(map(int, input().split())))

meetings.sort(key = lambda x: x[0])
meetings.sort(key = lambda x: x[1])
result = 1
last_element = 0

for i in range(1,n):
    if(meetings[last_element][1] <= meetings[i][0]):
        result += 1
        last_element = i
        
print(result)

File: 백준/Silver/1931. 회의실 배정/회의실 배정.js
const readline = require('readline').createInterface({
  input: process.stdin,
  output: process.stdout,
});

let input = [];

readline.on('line', function(line) {
  input.push(line)
}).on('close', function () {
  const n = Number(input.shift())
  const timeTable = input.map((row) => { 
    return row.split(" ").map(Number)
  })
  const answer = solution(n, timeTable)
  console.log(answer)
});


function solution(n, timeTable) {
  timeTable.sort(([s1,e1], [s2, e2]) => {
    if (s1 === s2) 
      return e1 - e2
    return s1 - s2
  })
  
  let answer = 0
  let endPoint = 0
  for (const [s,e] of timeTable) {
    if (endPoint <= s) {
      endPoint = e
      answer++
    }
    if (e < endPoint) {
      endPoint = e
    }
  }
  return answer
}